peerscout/keyword_extract/keyword_extract.py

```diff
@@ -192,74 +247,65 @@ def add_timestamp(record_list, timestamp_field_name, timestamp_as_string):
 
 
 def get_latest_state(
-        record_status_timestamp_list,
+        record_list,
+        timestamp_field_name: str = None
 ):
-    latest_timestamp = datetime.datetime.strptime(
-        "1900-01-10 00:00:00+0000",
-        ETL_STATE_TIMESTAMP_FORMAT
-    )
-    for record_status_timestamp in record_status_timestamp_list:
+    latest_timestamp = None
+    if timestamp_field_name:
+        timestamp_list = [
+            record.get(timestamp_field_name)
+            for record in record_list
+            if record.get(timestamp_field_name)
+        ]
         latest_timestamp = (
-            latest_timestamp
-            if latest_timestamp > record_status_timestamp
-            else record_status_timestamp
+            max(timestamp_list) if timestamp_list else None
         )
     return latest_timestamp
 
 
-def remove_cols_load_to_db_get_latest_timestamp(
-        json_list, full_temp_file_location,
+def write_to_jsonl_file(
+        data_with_extracted_keywords,
+        full_temp_file_location,
         keyword_extract_config
 ):
-    records_count = 0
-    write_disposition = (
-        WriteDisposition.WRITE_APPEND
-        if keyword_extract_config.table_write_append
-        else WriteDisposition.WRITE_TRUNCATE
-    )
-    with open(full_temp_file_location, "a") as write_file:
-        for record in json_list:
-            records_count += 1
-            if records_count % 1000 == 0:
-                load_data_to_bq(
-                    keyword_extract_config,
-                    full_temp_file_location,
-                    write_disposition
-                )
-                write_file.truncate(0)
-                LOGGER.info("%s records processed", str(records_count))
-            latest_timestamp = record.pop(
-                keyword_extract_config.state_timestamp_field
-            )
-            record.pop(keyword_extract_config.existing_keywords_field)
-            record.pop(keyword_extract_config.text_field)
+    with open(full_temp_file_location, "w") as write_file:
+        for record in data_with_extracted_keywords:
+            record.pop(keyword_extract_config.existing_keywords_field, None)
+            record.pop(keyword_extract_config.text_field, None)
+            record.pop(keyword_extract_config.state_timestamp_field, None)
             record = {key: value for key, value in record.items() if value}
             write_file.write(json.dumps(record, ensure_ascii=False))
             write_file.write("\n")
 
-            yield latest_timestamp
 
-        load_data_to_bq(
-            keyword_extract_config,
-            full_temp_file_location,
-            write_disposition
-        )
-        LOGGER.info("%s records processed", str(records_count))
+def iter_get_batches(iterable, size):
+    iterable = iter(iterable)
```

Comment:
minor: to me this is an _iterator_ (whereas the passed in value is an _iterable_)